src/Symfony/Component/DependencyInjection/LazyProxy/PhpDumper/DumperInterface.php

```diff
@@ -38,7 +38,7 @@ public function isProxyCandidate(Definition $definition);
      *
      * @return string
      */
-    public function getProxyFactoryCode(Definition $definition, $id, $methodName);
+    public function getProxyFactoryCode(Definition $definition, $id, $methodName = null);
```

Comment:
Well, I mean, we can make the argument required in 4.0. It will not forbid making 3.4 code compatible with 4.x (you need to make the argument optional in your implementation to support the 3.4 interface, but this does not forbid 4.x to make it required in the interface)